test/staticvector_test.cpp

```diff
@@ -0,0 +1,185 @@
+#include <cstdlib>
+#include <gtest/gtest.h>
+
+#include "engine/random.hpp"
+#include "utils/static_vector.hpp"
+
+using namespace devilution;
+
+namespace {
+
+constexpr int MAX_SIZE = 32;
+
+TEST(StaticVector, StaticVector_push_back)
+{
+	StaticVector<size_t, MAX_SIZE> container;
+
+	SetRndSeed(testing::UnitTest::GetInstance()->random_seed());
+	size_t size = RandomIntBetween(10, MAX_SIZE, true);
+
+	for (size_t i = 0; i < size; i++) {
+		container.push_back(i);
+	}
+
+	EXPECT_EQ(container.size(), size);
+	for (size_t i = 0; i < size; i++) {
+		EXPECT_EQ(container[i], i);
+	}
+}
+
+TEST(StaticVector, StaticVector_push_back_full)
+{
+	StaticVector<size_t, MAX_SIZE> container;
+
+	size_t size = MAX_SIZE;
+	for (size_t i = 0; i < size; i++) {
+		container.push_back(i);
+	}
+
+	EXPECT_EQ(container.size(), MAX_SIZE);
+	for (size_t i = 0; i < size; i++) {
+		EXPECT_EQ(container[i], i);
+	}
+
+#ifdef _DEBUG
+	ASSERT_DEATH(container.push_back(size + 1), "");
+#endif
+}
+
+TEST(StaticVector, StaticVector_erase)
+{
+	StaticVector<size_t, MAX_SIZE> container;
+	std::vector<size_t> expected;
+
+	SetRndSeed(testing::UnitTest::GetInstance()->random_seed());
+
+#ifdef _DEBUG
+	ASSERT_DEATH(container.erase(container.begin()), "");
+#endif
+
+	container = { 0, 1, 2, 3, 4, 5, 6, 7, 8, 9 };
+	expected = { 1, 2, 3, 4, 5, 6, 7, 8, 9 };
+	container.erase(container.begin());
+	EXPECT_EQ(container.size(), expected.size());
+	for (size_t i = 0; i < container.size(); i++) {
+		EXPECT_EQ(container[i], expected[i]);
+	}
+
+	expected = { 1, 2, 3, 4, 5, 6, 7, 8 };
+	container.erase(container.end() - 1, container.end());
+	EXPECT_EQ(container.size(), expected.size());
+	for (size_t i = 0; i < container.size(); i++) {
+		EXPECT_EQ(container[i], expected[i]);
+	}
+
+	size_t erasures = container.size();
+	for (size_t i = 0; i < erasures && container.size() > 0; i++) {

```

Comment:
nit: you never use `i` in this loop, this could simply be `while (!container.empty())` (also removes the need for the `erasures` constant).